File: server/src/Voting.Core/Entities/Comment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Voting.Core.Entities
{
    public class Comment
    {
        public int Id { get; set; }
        public string Message { get; set; }
        public DateTime PublishDate { get; set; }
        public bool IsBlocked { get; set; }

        public int? CommentId { get; set; }
        public IEnumerable<Comment> Comments { get; set; }

        public int QuestionId { get; set; }
        public Question Question { get; set; }

        public string RecipientId { get; set; }
        [ForeignKey("RecipientId")]
        public User Recipient { get; set; }

        public string UserId { get; set; }
        [ForeignKey("UserId")]
        [InverseProperty("Comments")]
        public User User { get; set; }
    }
}

File: client/Voting/src/app/core/models/Comment.ts
export class Comment{
    id: number;
    message: string;
    publishDate: Date;
    isMine: boolean;
    isBlocked: boolean;
    commentId: number;
    userId: string;
    userName: string;
    questionId: number;

    answersCount: number;
    comments: Comment[];
}